src/main/java/it/gov/pagopa/payhub/activities/activity/debtposition/ValidateDebtPositionActivityImpl.java

```diff
@@ -0,0 +1,132 @@
+            throw new ValidationException("Amount is not valid for this debt position type org");
+        }
+    }
+
+    private void validatePersonData(PersonDTO personDTO, DebtPositionTypeOrgDTO debtPositionTypeOrgDTO) {
+        if (personDTO == null) {
+            throw new ValidationException("The debtor is mandatory for installment");
+        }
+        if (StringUtils.isBlank(personDTO.getUniqueIdentifierCode())) {
+            throw new ValidationException("Unique identification code is mandatory");
+        }
+        if (!debtPositionTypeOrgDTO.isFlagAnonymousFiscalCode() && personDTO.getUniqueIdentifierCode().equals("ANONIMO")) {
+            throw new ValidationException("This organization installment type or installment does not allow an anonymous unique identification code");
+        }
+        if (StringUtils.isBlank(personDTO.getFullName())) {
+            throw new ValidationException("Beneficiary name is mandatory");
+        }
+        if (StringUtils.isBlank(personDTO.getEmail()) ||
+                !Utilities.isValidEmail(personDTO.getEmail())) {
+            throw new ValidationException("Email is not valid");
+        }
+    }
+
+    private void validateTransfers(List<TransferDTO> transferDTOList) {
+        if(CollectionUtils.isEmpty(transferDTOList)){
+            throw new ValidationException("At least one transfer is mandatory for installment");
+        }
+
+        if (transferDTOList.size() > 1) {
+            TransferDTO transferSecondaryBeneficiary = transferDTOList.stream()
+                    .filter(transfer -> (transfer.getTransferIndex() == 2)).findAny()
+                    .orElseThrow(() -> new ValidationException("Mismatch with transfers list"));
+
+            if (StringUtils.isBlank(transferSecondaryBeneficiary.getOrgFiscalCode()) ||
+                    !isValidPIVA(transferSecondaryBeneficiary.getOrgFiscalCode())) {
+                throw new ValidationException("Fiscal code of secondary beneficiary is not valid");
+            }
+            if (!isValidIban(transferSecondaryBeneficiary.getIban())) {
+                throw new ValidationException("Iban of secondary beneficiary is not valid");
+            }
+            checkTaxonomyCategory(transferSecondaryBeneficiary.getCategory());
+
+            if (transferSecondaryBeneficiary.getAmount() == null || transferSecondaryBeneficiary.getAmount() < 0) {
+                throw new ValidationException("The amount of secondary beneficiary is not valid");
+            }
+        }
+    }
+
+    private void checkTaxonomyCategory(String category) {
+        if (StringUtils.isBlank(category)) {
+            throw new ValidationException("Category of secondary beneficiary is mandatory");
+        } else {
+            String categoryCode = StringUtils.substringBeforeLast(category, "/") + "/";
+            Boolean categoryCodeExists = taxonomyDao.verifyCategory(categoryCode);
```

Comment:
what table is supposed to be queries to perform this check?

Reply:
mygov_tassonomia